sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -175,6 +175,30 @@ namespace Azure { namespace Core { namespace Http {
     NetworkAuthenticationRequired = 511, ///< HTTP 511 Network Authentication Required.
   };
 
+  /**
+   * @brief Defines a range of bytes within an HTTP resource, starting at an `Offset` and ending at
+   * `Offset + Length - 1` inclusively.
+   *
+   */
+  struct Range
+  {
+    /**
+     * @brief The starting point of the HTTP Range.
+     *
+     */
+    int64_t Offset = 0;
+
+    /**
+     * @brief The size of the HTTP Range.
+     *
+     */
+    int64_t Length = 0;
```

Comment:
Should `Length` instead default to the "gimme everything" sentinel value, so `Range{42}` (omitting the length) produces `bytes=42-` (also omitting the length)?

Should this class have a `static constexpr` member to name the "gimme everything" sentinel value?

Reply:
Yep, we do want `bytes=42-` when length is omitted, and not `bytes=42-0`. We can solve this by making length a `Nullable<int64_t>`, and we should do that :)